migration.md

```diff
@@ -0,0 +1,267 @@
+## Migration Guide From v4.X to v5.X
```

Comment:
I think the migration guide should be split into what users **have to do** and what is optional.

Reply:
The migration guide should be linked to from the top of the readme.

Reply:
For the two first optional migrations, we should explicitly point out that running `Defaults.migrate` is not needed.